controllers/argocd/applicationset.go

```diff
@@ -135,7 +136,10 @@ func (r *ReconcileArgoCD) reconcileApplicationSetDeployment(cr *argoprojv1a1.Arg
 	}
 
 	if existing := newDeploymentWithSuffix("applicationset-controller", "controller", cr); argoutil.IsObjectFound(r.Client, cr.Namespace, existing.Name, existing) {
-
+		if cr.Spec.ApplicationSet == nil {
+			log.Info(fmt.Sprintf("Deleting Deployment %s as applicationset is disabled", existing.Name))
+			return r.Client.Delete(context.TODO(), existing)
```

Comment:
When I removed the applicationSet entry from Argo CD CR, controller crashed with the below error. I edited the Argo CD CR to remove the `.spec.applicationSet` entry.

```
1.6542400631086211e+09	INFO	controller_argocd	reconciling ApplicationSet deployment
E0603 12:37:43.108966   18003 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 506 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x237e8a0?, 0x3477f10})
	/Users/aveerama/go/src/github.com/iam-veeramalla/argocd-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x86
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x8?})
	/Users/aveerama/go/src/github.com/iam-veeramalla/argocd-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x75
panic({0x237e8a0, 0x3477f10})
	/usr/local/go/src/runtime/panic.go:838 +0x207
github.com/argoproj-labs/argocd-operator/controllers/argocd.getArgoApplicationSetCommand(0xc000b80000)
	/Users/aveerama/go/src/github.com/iam-veeramalla/argocd-operator/controllers/argocd/applicationset.go:47 +0x285
github.com/argoproj-labs/argocd-operator/controllers/argocd.applicationSetContainer(_)
	/Users/aveerama/go/src/github.com/iam-veeramalla/argocd-operator/controllers/argocd/applicationset.go:190 +0x152
```